core/src/test/java/org/infinispan/persistence/BaseNonBlockingStoreTest.java

```diff
@@ -0,0 +1,660 @@
+import org.infinispan.container.impl.InternalEntryFactory;
+import org.infinispan.container.impl.InternalEntryFactoryImpl;
+import org.infinispan.distribution.ch.KeyPartitioner;
+import org.infinispan.marshall.TestObjectStreamMarshaller;
+import org.infinispan.marshall.persistence.PersistenceMarshaller;
+import org.infinispan.marshall.persistence.impl.MarshalledEntryUtil;
+import org.infinispan.persistence.spi.CacheLoader;
+import org.infinispan.persistence.spi.CacheWriter;
+import org.infinispan.persistence.spi.InitializationContext;
+import org.infinispan.persistence.spi.MarshallableEntry;
+import org.infinispan.persistence.spi.NonBlockingStore;
+import org.infinispan.persistence.spi.PersistenceException;
+import org.infinispan.persistence.support.EnsureNonBlockingStore;
+import org.infinispan.persistence.support.NonBlockingStoreAdapter;
+import org.infinispan.persistence.support.WaitNonBlockingStore;
+import org.infinispan.protostream.ProtobufUtil;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.protostream.SerializationContextInitializer;
+import org.infinispan.test.AbstractInfinispanTest;
+import org.infinispan.test.TestDataSCI;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.test.data.Key;
+import org.infinispan.test.data.Person;
+import org.infinispan.test.fwk.NonBlockingTest;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.infinispan.test.fwk.TestInternalCacheEntryFactory;
+import org.infinispan.util.ControlledTimeService;
+import org.infinispan.util.PersistenceMockUtil;
+import org.infinispan.util.concurrent.CompletionStages;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import io.reactivex.rxjava3.core.Flowable;
+
+/**
+ * This is a base class containing various unit tests for each and every different CacheStore implementations. If you
+ * need to add Cache/CacheManager tests that need to be run for each cache store/loader implementation, then use
+ * BaseStoreFunctionalTest.
+ */
+// this needs to be here for the test to run in an IDE
+@Test(groups = "unit", testName = "persistence.BaseNonBlockingStoreTest")
+@NonBlockingTest
+public abstract class BaseNonBlockingStoreTest extends AbstractInfinispanTest {
+
+   protected static final int WRITE_DELETE_BATCH_MIN_ENTRIES = 80;
+   protected static final int WRITE_DELETE_BATCH_MAX_ENTRIES = 120;
+   protected TestObjectStreamMarshaller marshaller;
+
+   protected WaitNonBlockingStore<Object, Object> store;
+   protected ControlledTimeService timeService;
+   private InternalEntryFactory factory;
+   private final IntSet segments = IntSets.immutableSet(0);
+
+   protected static <K, V> NonBlockingStore<K, V> asNonBlockingStore(CacheLoader<K, V> loader) {
+      return new NonBlockingStoreAdapter<>(loader);
+   }
+
+   protected static <K, V> NonBlockingStore<K, V> asNonBlockingStore(CacheWriter<K, V> writer) {
```

Comment:
Neither `asNonBlockingStore` method are ever called.

Reply:
This is for future. My thought was to convert the previous BaseStoreTest to extend this one and just use these methods to convert. I will try to squeeze that in here possibl. This way we don't have 2 base test classes doing essentially the same thing.